### DISCUSSION TITLE
Game crashes upon trying to break or place a block

### DISCUSSION THREAD

**Discussion by She-leap** [FabricMC/fabric-api#1906] [Player support]
For some reason, upon trying to break or place a block or trying to attack in 3rd or first person crashes the game. Minecraft throws a java.lang.NullPointerException: Cannot invoke "java.util.Optional.isPresent()" because "armPose" is null error which is weird as I never encountered something like this.
Crash Log below:
https://pastebin.com/H59NQhvA

**Comment by warjort** [comment]
````
java.lang.NullPointerException: Cannot invoke "java.util.Optional.isPresent()" because "armPose" is null
	at net.minecraft.class_1007.handler$deb001$getArmPose(class_1007.java:4025)
	at net.minecraft.class_1007.method_4210(class_1007.java)
````

That ````getArmPose```` is from a mod doing a mixin.
Sadly, the mod doesn't give the mixin a useful name so we don't know which mod it is.

You can try installing the "not enough crashes" mod to see if that tells you which mod is causing the problem.
Otherwise, you will have to remove mods until you find the problem one.

I can tell you that class_1007 is the "PlayerEntityRenderer" so the mod will be doing something with how the player is drawn/animated.
That might help to narrow it down?

**Reply by She-leap** [reply]
https://pastebin.com/VTAPaJ80
Crash log with not enough crashes still displays a similar "armpose" is null error. NEC doesn't really tell me which mod is causing it either.

**Comment by warjort** [comment]
I did some searching on github and can't find the mod that does the getArmPose mixin.
But there is another mixin in that stacktrace:
````
at net.minecraft.client.render.GameRenderer.redirect$cjc000$disableVanillaHandRendering(GameRenderer:5048)
````
which is coming from this iris shader mod:
https://github.com/IrisShaders/Iris/blob/18d4ce87efbd63253f2b98a989bc1e8b486c594d/src/main/java/net/coderbot/iris/mixin/MixinGameRenderer.java#L45

I doubt that is the cause of your problem though, since from the stacktrace, it looks like it is just following the normal vanilla logic.

Like I said above, the only thing I can suggest is to experiment with removing rendering related mods until you find the mod causing the problem.
